sandbox/cgroup/monitor.go

```diff
@@ -54,6 +50,9 @@ type inotifyWatcher struct {
 	// is decremented, and when it reaches zero, it is removed and inotify is
 	// notified to stop monitoring it.
 	pathList map[string]int32
+
+	// serialize operations
+	mu sync.Mutex
```

Comment:
Indeed it does, I tweaked this quite a bit now I'm sure it needs some more tweaks, I'm still not super happy but it's safer now I feel. The alternative would be to move back to a full channel based approach instead of the mutex but that comes with it's own questions.

Reply:
I think that would be simpler since we'd only have one synchronization method instead of two. However, I wonder if that would introduce a race condition between adding paths to monitor and stopping the main loop. As in, we'd push some path to be added, the inotify channel would close before that was read, causing the main loop to stop, and then the MonitorDelete would just be blocked. What are your concerns with it?